### PR TITLE
`warp` replace `httpmock`

### LINKED ISSUE #1207: Support dynamic response in `mock-server`
### Context
When we try to generate different challenge-codes each time in mock mode , this requires mock-server to return different responses according to different challenge-codes.

Maybe the [httpmock](https://github.com/alexliesenfeld/httpmock) doesn't support dynamic response. We can use normal http library instead, like [warp](https://github.com/seanmonstar/warp)

### PR DESCRIPTION
**Pull request by ziming-zung**
This PR resolves #1207

### CODE REVIEW

File: tee-worker/litentry/core/data-providers/src/graphql.rs [hunk 1]
```diff
@@ -97,26 +96,11 @@ impl VerifiedCredentialsIsHodlerIn {
...
-		if self.token_address.is_empty().not() {
-			flat += &format!(",tokenAddress:\"{}\"", &self.token_address.clone());
-		}
-		flat += &format!(",minimumBalance:{:?}", self.min_balance.clone());
-		flat
+	pub fn to_graphql(&self) -> String {
+		let addresses_str = format!("{:?}", self.addresses);
+		let network = format!("{:?}", self.network).to_lowercase();
+		let q = format!("VerifiedCredentialsIsHodler(addresses:{}, fromDate:\"{}\", network:{}, tokenAddress:\"{}\",minimumBalance:{:?}){{isHodler,address}}", addresses_str, self.from_date, network, self.token_address, self.min_balance);
+		format!("{{{}}}", q)
```
Comment (DogLi) [inline tee-worker/litentry/core/data-providers/src/graphql.rs]: format twice, making them into one is better.
```
format!(r#"{{VerifiedCredentialsIsHodler.....{{isHodler,address}}}}"#)
```

**Comment by DogLi** [comment]
LGFM

File: tee-worker/litentry/core/mock-server/Cargo.toml [hunk 2]
```diff
@@ -5,17 +5,18 @@ version = "0.1.0"
...
-tokio = { version = "1.5", features = ["sync", "macros", "rt-multi-thread", "signal"] }
+log = "0.4"
+serde_json = { version = "1.0", features = ["alloc"] }
+tokio = { version = "1.5", features = ["full"] }
+warp = "0.3"
 
-codec = { package = "parity-scale-codec", version = "3.0.0", default-features = false, features = ["derive"] }
-litentry-primitives = { path = "../../primitives", default-features = false }
-sp-core = { git = "https://github.com/paritytech/substrate", branch = "polkadot-v0.9.36", default-features = false }
+# Subtrate
```
Comment (Kailai-Wang) [inline tee-worker/litentry/core/mock-server/Cargo.toml]: A minor typo: "Substrate"

**Review by Kailai-Wang — APPROVED** [review]
Looks good, thanks!

File: tee-worker/litentry/core/data-providers/src/graphql.rs [hunk 3]
```diff
@@ -97,26 +96,11 @@ impl VerifiedCredentialsIsHodlerIn {
...
-		}
-		if self.token_address.is_empty().not() {
-			flat += &format!(",tokenAddress:\"{}\"", &self.token_address.clone());
-		}
-		flat += &format!(",minimumBalance:{:?}", self.min_balance.clone());
-		flat
+	pub fn to_graphql(&self) -> String {
+		let addresses_str = format!("{:?}", self.addresses);
+		let network = format!("{:?}", self.network).to_lowercase();
+		let q = format!("VerifiedCredentialsIsHodler(addresses:{}, fromDate:\"{}\", network:{}, tokenAddress:\"{}\",minimumBalance:{:?}){{isHodler,address}}", addresses_str, self.from_date, network, self.token_address, self.min_balance);
```
Comment (BillyWooo) [inline tee-worker/litentry/core/data-providers/src/graphql.rs]: Have you double checked that "token_address is empty"?

File: tee-worker/litentry/core/data-providers/src/graphql.rs (hunk 1 above)
Comment (ziming-zung) [reply tee-worker/litentry/core/data-providers/src/graphql.rs]: Have corrected it

File: tee-worker/litentry/core/data-providers/src/graphql.rs (hunk 3 above)
Comment (ziming-zung) [reply tee-worker/litentry/core/data-providers/src/graphql.rs]: Have fixed it, pls check.

**Review by BillyWooo — APPROVED** [review]
LGTM

**Pull request merged**